src/EditorFeatures/Next/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -104,15 +110,34 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
                 // We don't have anything cached, so we'll just get it now lazily and not hold onto it. The workspace layer will ensure
                 // that we maintain snapshot rules for the document options. We'll also run it on the thread pool
                 // as in some builds the ICodingConventionsManager captures the thread pool.
-                var conventionsAsync = Task.Run(() => _codingConventionsManager.GetConventionContextAsync(path, cancellationToken));
+                var conventionsAsync = Task.Run(() => GetConventionContextAsync(path, cancellationToken));
+                var context = await conventionsAsync.ConfigureAwait(false);
+                if (context == null)
```

Comment:
IIRC you can still do this inside the using, which I think I would prefer so there's no chance of somebody else putting a return in the middle.